airflow/datasets/__init__.py

```diff
@@ -159,6 +165,28 @@ def iter_datasets(self) -> Iterator[tuple[str, Dataset]]:
         raise NotImplementedError
 
 
+@attr.define()
+class DatasetAlias(BaseDataset):
+    """A represeation of dataset alias which is used to create dataset during the runtime."""
+
+    name: str
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, DatasetAlias):
+            return self.name == other.name
+        return NotImplemented
+
+    def __hash__(self) -> int:
+        return hash(self.name)
```

Comment:
Does setting `eq=True` in `define` work? Also I think attrs provide a `__hash__`, not sure if that’s good.

Reply:
> Does setting eq=True in define work? 

According to https://www.attrs.org/en/stable/hashing.html, `it's almost certainly a bug.`

> attrs provide a __hash__,

`attrs will happily write a __hash__ method for you [[1]](https://www.attrs.org/en/stable/hashing.html#fn1), however it will not do so by default.`

---

I also did a quick test on `frozen`, but didn't seem to work.

Reply:
the original way looks better to me. let me know if we still have concern on this. Thanks!